### PR TITLE
SUP-32804 support stopProcessingOnError in MR for distribute and expo…

### PR DESCRIPTION
**Pull request by salma2222**
…rt batch jobs

### CODE REVIEW

File: plugins/scheduled_task/lib/MediaRepurposingHandler.php [hunk 1]
```diff
@@ -69,6 +69,91 @@ public function shouldConsumeDataChangedEvent(BaseObject $object, $previousVersi
...
+
+		if($isDistributeTask || $isExportTask)
+		{
+			return true;
+		}
+
+		return false;
+	}
+
+	private function getPostProcessMRMetadata($mrMetadata)
```
Comment (inbal-ben-david) [inline plugins/scheduled_task/lib/MediaRepurposingHandler.php:150]: the name misleading, you don't just get, you also update so would give naming like getUpdated..

File: plugins/scheduled_task/lib/processors/KGenericProcessor.php [hunk 2]
```diff
@@ -48,6 +48,28 @@ protected function additionalActions($profile, $object, $tasksCompleted, $error)
...
 
+	protected function shouldProcessTask($object, $objectTask)
+	{
+		return $object instanceof KalturaBaseEntry && $objectTask->stopProcessingOnError
+			&& in_array($objectTask->type, $this->getSupportedTypesForProcess()) && $this->isPartnerSupportedForProcess($object->partnerId);
+	}
+
+	protected function getSupportedTypesForProcess()
+	{
+		$distributeTaskType = ScheduledTaskContentDistributionPlugin::getApiValue('Distribute');
```
Comment (inbal-ben-david) [inline plugins/scheduled_task/lib/processors/KGenericProcessor.php:59]: dont we have an enum for distribute

File: plugins/scheduled_task/lib/processors/KMediaRepurposingProcessor.php [hunk 3]
```diff
@@ -81,22 +81,29 @@ private function updateMetadataStatusForMediaRepurposing(KalturaScheduledTaskPro
 		$metadata = $this->getMetadataOnObject($object->id, $metadataProfileId);
 
 		$xml = ($metadata && $metadata->xml) ? $metadata->xml : null;
+		$process = $this->createProcessField($object, $profile->objectTasks[0]);
```
Comment (inbal-ben-david) [inline plugins/scheduled_task/lib/processors/KMediaRepurposingProcessor.php:84]: will we always reach here with objectTasks[0]?
can it be null?

File: plugins/scheduled_task/lib/processors/KMediaRepurposingProcessor.php [hunk 4]
```diff
@@ -122,6 +129,36 @@ private function getMetadataOnObject($objectId, $metadataProfileId)
 		return null;
 	}
 
+	private function createProcessField($object, $task)
```
Comment (inbal-ben-david) [inline plugins/scheduled_task/lib/processors/KMediaRepurposingProcessor.php:132]: protected

File: plugins/scheduled_task/lib/processors/KMediaRepurposingProcessor.php [hunk 5]
```diff
@@ -122,6 +129,36 @@ private function getMetadataOnObject($objectId, $metadataProfileId)
...
+	{
+		$distributeTaskType = ScheduledTaskContentDistributionPlugin::getApiValue('Distribute');
```
Comment (inbal-ben-david) [inline plugins/scheduled_task/lib/processors/KMediaRepurposingProcessor.php:134]: enum?

**Review by inbal-ben-david — COMMENTED** [review]
also add release notes regarding the required config

File: plugins/scheduled_task/lib/processors/KGenericProcessor.php (hunk 2 above)
Comment (salma2222) [reply plugins/scheduled_task/lib/processors/KGenericProcessor.php:59]: not working - Interface 'IKalturaPluginEnum' not found

File: plugins/scheduled_task/lib/processors/KMediaRepurposingProcessor.php (hunk 5 above)
Comment (salma2222) [reply plugins/scheduled_task/lib/processors/KMediaRepurposingProcessor.php:138]: Interface 'IKalturaPluginEnum' not found

**Review by inbal-ben-david — COMMENTED** [review]
add release notes

**Pull request merged**